docs/gke/configs/cluster.jinja

```diff
@@ -72,59 +83,85 @@ resources:
       monitoringService: monitoring.googleapis.com/kubernetes
       {% else %}
       initialClusterVersion: 1.9.6-gke.1
-      {% endif %}      
+      {% endif %}
+      podSecurityPolicyConfig:
+        enabled: {{ properties['podSecurityPolicy'] }}
+      {% if properties['privatecluster'] %}
+      ipAllocationPolicy:
+        createSubnetwork: true
+        useIpAliases: true
+      masterIpv4CidrBlock: {{ properties['masterIpv4CidrBlock'] }}
+      privateCluster: true
+      masterAuthorizedNetworksConfig: {{ properties['masterAuthorizedNetworksConfig'] }}
+      {% endif %}
       nodeConfig:
         machineType: n1-standard-1
+        serviceAccount: {{ properties['serviceAccountName'] }}@{{ env['project'] }}.iam.gserviceaccount.com
```

Comment:
Do we want to use the service account as the VM service account? I think a better practice is to move in the direction of creating secrets containing service account keys. This way we can provide finer grained permissions then what we get from node service accounts.

Reply:
This service account will not be used by pods. This service account is intended to replace the default compute engine service account which has Project Editor Role. We will be using secrets and service account keys to give access to individual pods.